src/containers/BooksForm.jsx

```diff
@@ -1,17 +1,80 @@
 import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { createBook } from '../actions/index';
+import categories from '../data/categories';
 
-const categories = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
-
-const BookForm = () => (
-  <div>
-    <form action="">
-      <input type="text" placeholder="Title" />
-      <select name="category" id="cat">
-        {categories.map(cat => (<option key={cat} value={cat}>{cat}</option>))}
-      </select>
-      <input type="submit" value="Save" />
-    </form>
-  </div>
-);
-
-export default BookForm;
+class BooksForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      id: Math.floor(Math.random() * 1000),
+      title: '',
+      category: categories[0],
+    };
+  }
+
+  handleChange = (e) => {
+    if (e.target.name === 'title') {
+      this.setState({
+        title: e.target.value,
+      });
+    } else if (e.target.name === 'category') {
+      this.setState({
+        category: e.target.value,
+      });
+    }
+  }
+
```

Comment:
alternatively too,
```
handleChange(event) {
  const name = event.target.name;

  this.setState({
    title: event.target.value
    [name]: event.target.value
  })
}


```

Reply:
Thanks, @shloch I'm reading your links and figured out how to solve it